content/courses/native-onchain-development/program-derived-addresses.md

```diff
@@ -22,258 +22,353 @@ description: "Get a deeper understanding of PDAs."
-play an important role in how we use PDAs for data storage and retrieval.
+Program Derived Addresses (PDAs) are addresses that, instead of being public
+keys, are calculated (or 'found') based on a combination of:
+
+- The program ID
+- A set of "seeds" determined by the programmer.
+
+More on this later, but these seeds will play a role in using PDAs for data
+storage and retrieval.
 
 PDAs serve two main functions:
 
 1. Provide a deterministic way to find a given item of data for a program
-2. Authorize the program from which a PDA was derived to sign on its behalf in
-   the same way a user may sign with their secret key
+2. Authorize the program that owns a PDA to sign on the PDAs behalf, just like a
+   user signs for their own account using their secret key.
 
-In this lesson we'll focus on using PDAs to find and store data. We'll discuss
-signing with a PDA more thoroughly in a future lesson where we cover Cross
-Program Invocations (CPIs).
+This lesson will focus on using PDAs to find and store data. We'll discuss
+signing with a PDA more thoroughly in a future lesson, where we will cover
+Cross-Program Invocations (CPIs).
 
 ### Finding PDAs
 
-PDAs are not technically created. Rather, they are _found_ or _derived_ based on
-a program ID and one or more input seeds.
+Technically, PDAs are _found_ or _derived_ based on a program ID and one or more
+input seeds.
 
-Solana keypairs can be found on what is called the Ed25519 Elliptic Curve
-(Ed25519). Ed25519 is a deterministic signature scheme that Solana uses to
-generate corresponding public and secret keys. Together, we call these keypairs.
+Unlike other Solana accounts, PDAs are not public keys and don't have secret
+keys. Since public keys are on Solana's Ed25519 curve, PDAs are sometimes called
+'off curve addresses'.
 
-Alternatively, PDAs are addresses that lie _off_ the Ed25519 curve. This means
-PDAs are not public keys, and don't have private keys. This property of PDAs is
-essential for programs to be able to sign on their behalf, but we'll cover that
-in a future lesson.
+PDAs are found using a hashing function that deterministically generates a PDA
+using the program ID and seeds. Both Solana frontend and backend code can
+determine an address using the program ID and seeds, and the same program with
+the same seeds always results in the same Program Derived Address.
 
-To find a PDA within a Solana program, we'll use the `find_program_address`
-function. This function takes an optional list of “seeds” and a program ID as
-inputs, and then returns the PDA and a bump seed.
+### Seeds
 
-```rust
-let (pda, bump_seed) = Pubkey::find_program_address(&[user.key.as_ref(), user_input.as_bytes().as_ref(), "SEED".as_bytes()], program_id)
-```
+“Seeds” are inputs in the `find_program_address` function, this method provides
+an additional seed called a “bump seed.” The find*program_address method adds a
+numeric seed called a bump seed that ensures the result is \_off* on the Ed25519
```

Comment:
I have just noticed it was an issue with my grammarly configuration changing `""` to `”“` and `'` to `’`  because the markdown file is using both. I will  use `""`  and `'` for consistency. If that's not okay let me know.